u7-components/src/views/BookDetails.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <div class="book-details">
+    <img :src="`/images/books/${book.isbn13}.jpg`" :alt="book.title" />
+    <p>{{ book.description }}</p>
+
+    <h2>You might also like...</h2>
+    <div class="related">
+      <BookCard v-for="book in relatedBooks" :key="`book-related-${book.isbn13}`" :book="book"/>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import type { Route, NavigationGuardNext } from 'vue-router'
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Book, getBook, getRelatedBooks } from '@/_services/fakeapi'
+import BookCard from './BookCard.vue'
+
+Component.registerHooks([
+  'beforeRouteEnter',
+  'beforeRouteLeave',
+  'beforeRouteUpdate'
+])
+
+@Component({ components: { BookCard } })
+export default class BookDetails extends Vue {
+    book: Book | null = null
+    relatedBooks: Book[] = []
+
+    created (): void {
+      this.loadData(this.$route.params.isbn13)
+    }
+
+    beforeRouteUpdate (to: Route, _from: Route, next: NavigationGuardNext): void {
+      this.loadData(to.params.isbn13)
+      next()
+    }
+
+    loadData (isbn13: string): void {
+      this.book = getBook(isbn13)
+      this.relatedBooks = getRelatedBooks(this.book)
+    }
```

Comment:
ตัวแบ่งบรรทัดที่ดี มันทำให้อ่านง่ายขึ้นมาก!
Good line-breaks. It makes it much easier to read!